lib/project.rb

```diff
@@ -157,31 +157,50 @@ def check_run_termination
       @github.handle_cancel
     end
 
-    def init_workspace
-      unless @options.common.workspace_path.nil?
-        @workspace_path = @options.common.workspace_path
-        return
-      end
+    def session
+      @workspace_path = @options.test.session
+
+      raise AutoHCKError, 'Workspace path does not exist could not load session' unless File.directory?(@workspace_path)
 
+      @logger.info("Loading session from #{@workspace_path}")
+    end
+
+    def create_session
       @workspace_path = File.join(@config['workspace_path'],
                                   @engine_name,
                                   @engine_tag,
                                   @timestamp)
+
       begin
         FileUtils.mkdir_p(@workspace_path)
       rescue Errno::EEXIST
         @logger.warn('Workspace path already exists')
       end
       @logger.info("Workspace path: #{@workspace_path}")
 
+      Session.save(@workspace_path, @options)
+    end
+
+    def init_session
```

Comment:
This method should still be named `init_workspace`. It initialized more than `session.json`, which is part of workspace.